common/hash_data_adapter_hidden_files.go

```diff
@@ -66,9 +67,18 @@ func (a *HiddenFileDataAdapter) SetHashData(relativePath string, data *SyncHashD
 
 	metaFile := a.getHashPath(relativePath)
 
-	f, err := os.OpenFile(metaFile, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0644)
-	if err != nil {
-		return fmt.Errorf("failed to open hash meta file: %w", err)
+	var f *os.File
+	_, err := os.Stat(metaFile)
+	if os.IsNotExist(err) {
```

Comment:
Both if and else are doing almost the same thing, we shall rather unify the case. Only the log differs it seems here.

Reply:
There is one use case where it fails if we use a single OpenFile() API with os.O_CREATE|os.O_TRUNC|os.O_RDWR. 
Once the metafile is hidden and if we try to run the same azcopy command it fails to open the meta file with these flags being used combined os.O_CREATE|os.O_TRUNC|os.O_RDWR for which I have seperated the 2 calls.

Reply:
Added the comment in the code as well.